azkaban-common/src/main/java/azkaban/project/DirectoryYamlFlowLoader.java

```diff
@@ -100,32 +101,38 @@ public ValidationReport loadProjectFlow(final Project project, final File projec
   private void convertYamlFiles(final File projectDir) {
     // Todo jamiesjc: convert project yaml file.
 
-    final File[] flowFiles = projectDir.listFiles(new SuffixFilter(Constants.FLOW_FILE_SUFFIX));
-    for (final File file : flowFiles) {
+    for (final File file : projectDir.listFiles(new SuffixFilter(Constants.FLOW_FILE_SUFFIX))) {
       final NodeBeanLoader loader = new NodeBeanLoader();
       try {
         final NodeBean nodeBean = loader.load(file);
         final AzkabanFlow azkabanFlow = (AzkabanFlow) loader.toAzkabanNode(nodeBean);
-        final Flow flow = convertAzkabanFlowToFlow(azkabanFlow);
+        final Flow flow = convertAzkabanFlowToFlow(azkabanFlow, azkabanFlow.getName(), file);
         this.flowMap.put(flow.getId(), flow);
       } catch (final FileNotFoundException e) {
         logger.error("Error loading flow yaml files", e);
       }
     }
   }
 
-  private Flow convertAzkabanFlowToFlow(final AzkabanFlow azkabanFlow) {
-    final Flow flow = new Flow(azkabanFlow.getName());
+  private Flow convertAzkabanFlowToFlow(final AzkabanFlow azkabanFlow, final String flowName,
+      final File flowFile) {
+    final Flow flow = new Flow(flowName);
     final Props props = azkabanFlow.getProps();
     FlowLoaderUtils.addEmailPropsToFlow(flow, props);
+    props.setSource(flowFile.getName());
```

Comment:
should we set path instead of name as source?

Reply:
Name is enough because all the flow yaml files are under the same project directory. And we can retrieve the flow file from DB based on their names. We don't need to support subdirectories in the new design so there is no need to specify the path. Even today we store .job and .properties file name as the source instead of the path.